library/models/product.py

```diff
@@ -281,6 +328,49 @@ def _uniq(seq):
                         ('code_uniq', 'unique (code)',
                          'Code of the product must be unique !')]
 
+    @api.multi
+    def action_purchase_order(self):
+        purchase_line_obj = self.env['purchase.order.line']
+        purchase = purchase_line_obj.search([('product_id', '=', self.id)])
```

Comment:
I doubt this search will give you the desired result. Don't you think you need another filter?